src/payment/strategies/paypal-commerce/paypal-commerce-alternative-methods-strategy.ts

```diff
@@ -0,0 +1,145 @@
+import { Cart } from '../../../cart';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError, PaymentMethodInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { ApproveDataOptions,
+    ButtonsOptions,
+    PaypalCommerceCreditCardPaymentInitializeOptions,
+    PaypalCommerceInitializationData,
+    PaypalCommercePaymentInitializeOptions,
+    PaypalCommercePaymentProcessor, PaypalCommerceScriptParams,
+    PaypalCommerceSDKFunding } from './index';
+
+export default class PaypalCommerceAlternativeMethodsStrategy implements PaymentStrategy {
+    private _orderId?: string;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor
+    ) {}
+
+    async initialize({methodId, gatewayId, paypalcommerce}: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+
+        const multiOptionId = methodId;
+        let fundingKey: keyof PaypalCommerceSDKFunding;
+        if (gatewayId) {
+            methodId = gatewayId;
```

Comment:
Is there a reason why you're replacing `methodId` with `gatewayId`? I think they refer to different things. For multi-option providers, `gatewayId` refers to the name of a payment provider and `methodId` refers to the name of a payment method. To retrieve a payment method of a multi-option provider, you need both IDs, i.e.: `getPaymentMethodOrThrow(methodId, gatewayId)`

Reply:
removed file as it is no longer needed